code/modules/admin/verbs/selectequipment.dm

```diff
@@ -0,0 +1,197 @@
+	target = M
+
+/datum/select_equipment/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "SelectEquipment", "Select Equipment")
+		ui.open()
+		ui.set_autoupdate(FALSE)
+
+/datum/select_equipment/ui_state(mob/user)
+	return GLOB.admin_state
+
+/datum/select_equipment/ui_status(mob/user, datum/ui_state/state)
+	if(QDELETED(target))
+		return UI_CLOSE
+	return ..()
+
+/datum/select_equipment/ui_close(mob/user)
+	clear_human_dummy(dummy_key)
+	qdel(src)
+
+/datum/select_equipment/proc/init_dummy()
+	dummy_key = "selectequipmentUI_[target]"
+	dummy = generate_or_wait_for_human_dummy(dummy_key)
+	var/mob/living/carbon/C = target
+	if(istype(C))
+		C.dna.transfer_identity(dummy)
+		dummy.updateappearance()
+
+	unset_busy_human_dummy(dummy_key)
+	return
+
+/*
+ * category (string) - The tab it will be under
+ * path (typepath or string) - This will sent this back to ui_act to preview or spawn in an outfit
+ * name (string) - Will be the text on the button
+ * priority (int) - default 0, 1 for favorites, 2 for priority buttons
+*/
+#define OUTFIT_ENTRY(category, path, name, priority) list("category" = category, "path" = path, "name" = name, "priority" = priority)
+//passes a ref instead of path, every entry needs some sort of unique identifier
+//and GLOB.custom_outfits allows duplicate names
+#define OUTFIT_CUSTOM_ENTRY(category, ref, name, priority) list("category" = category, "ref" = ref, "name" = name, "priority" = priority)
+
+/datum/select_equipment/proc/make_outfit_entries(category="General", list/L)
+	var/list/entries = list()
+	for(var/path in L)
+		var/datum/outfit/O = path
+		var/priority = 0
+		if(path == /datum/outfit/job/roboticist) //dummy check here until I get favorites working
+			priority = 1
+		entries += list(OUTFIT_ENTRY(category, path, initial(O.name), priority))
+	return entries
+
+
+//GLOB.custom_outfits lists outfit *objects* so we'll need to do some custom handling for it
+/datum/select_equipment/proc/make_custom_outfit_entries(list/L)
+	var/list/entries = list()
+	for(var/datum/outfit/O in L)
+		entries += list(OUTFIT_CUSTOM_ENTRY("Custom", REF(O), O.name, 0)) //it's either this or special handling on the UI side
+	return entries
+
+/datum/select_equipment/ui_data(mob/user)
+	var/list/data = list()
+	if(!dummy)
+		init_dummy()
+
+	var/datum/preferences/prefs
+	if(target.client)
+		prefs = target.client.prefs
+	var/icon/dummysprite = get_flat_human_icon(null, prefs=prefs, dummy_key = dummy_key, outfit_override = selected_outfit)
+	data["icon64"] = icon2base64(dummysprite)
+	data["name"] = target
+
+	var/list/custom
+	custom += make_custom_outfit_entries(GLOB.custom_outfits)
+	data["custom_outfits"] = custom
+	data["current_outfit"] = selected_name
+	return data
+
+
+/datum/select_equipment/ui_static_data(mob/user)
+	var/list/data = list()
+	if(!cached_outfits)
+		cached_outfits = list()
+		cached_outfits += list(OUTFIT_ENTRY("General", /datum/outfit, "Naked", 2))
```

Comment:
Magic 2. Move it to a define.

Reply:
Ideally these magic values would be sent through to ui_static_data as well.

Reply:
No need to send them to the UI, they're just there [as priorities for sorting](https://github.com/tgstation/tgstation/pull/57673/files#diff-0931ef83d6077b6246c7d86ee4650436bea50aa7397bc02f5ba86145d5d9f49aR100)

Reply:
Nevermind; after I finish favorites, `priorities` will turn into a bool and the magic numbers will be gone